### PR TITLE
WebGL 1.0 spec update: no shader binaries in WebGL

### PR DESCRIPTION
**Pull request by Richard-Yunchao**
PTAL, @zhenyao and @kenrussell .

See the discussion at #2340.

### CODE REVIEW

**Comment by dmikis** [comment]
Anchor's missing. Also it there a need to specify this since there's simply no `shaderBinary` call in WebGL API?

**Comment by dmikis** [comment]
This bit is very unclear:

>  Shader compiler should be always true

I assume it refer to `GL_SHADER_COMPILER` enum, but, as with `shaderBinary` entry point, there's no such enum in WebGL.

**Comment by Richard-Yunchao** [comment]
@dmikis , thanks for your review. I think it is better to explicitly say this. It means that we deliberately don't want to expose shaderBinary to web developers, instead of we FORGET to expose the API. And that is also the reason why we have a standalone section to enumerate all differences between WebGL 1 and GLES 2.0.

Per the statements "Shader compiler should be always true", maybe we can remove this sentence, maybe we can have a better words to express this meaning. Any suggestions about this? @zhenyao and @kenrussell ?

File: specs/latest/1.0/index.html [hunk 1]
```diff
@@ -3481,6 +3481,14 @@ <h2 class="no-toc">Table of contents</h2>
 
 </p>
 
+    <h3>No Shader Binaries</h3>
+
+<p>
+
+Accessing binary representations of compiled shaders is not supported in the WebGL API. This includes OpenGL ES 2.0 ShaderBinary entry point. Shader compiler should be always true. In addition, querying shader binary formats and shader compiler with getparameter is not supported in the WebGL API.
```
Comment (zhenyao) [inline specs/latest/1.0/index.html]: Shader compiler is a separate issue from shader binaries.  Let's address it in a separate point (or a separate CL).

Add <code /> around ShaderBinary and getParameter (capital P)

**Comment by dmikis** [comment]
@Richard-Yunchao IMHO, it's obvious that it's not forgotten. "Differences Between WebGL and OpenGL ES 2.0" section is more about explicitly stating differences in behaviour between GLES entry points and corresponding WebGL ones. And it's clearly needed there since we refer to GLES man pages as description of entry points. Why we need to explicitly state that we omit some entry points isn't obvious for me, to be honest.

File: specs/latest/1.0/index.html (hunk 1 above)
Comment (Richard-Yunchao) [inline specs/latest/1.0/index.html]: Thanks for your review, @zhenyao . I have updated the pr according to your comments. Please take another look. Thanks a lot!

**Comment by Richard-Yunchao** [comment]
@dmikis , You know, shader binary is not only an API, but a standalone feature. we also remove a couple of GLenums like NUM_SHADER_BINARY_FORMAT and SHADER_BINARY_FORMAT from getParameter. 

If someone think the differences between WebGL and GLES is obvious, so we don't clearly state it in spec. IMHO, that is not a good way to draft a spec. I think we should clearly state ALL differences between WebGL and GLES. We should not let users to GUESS. Someone think some differences are obvious, the others may don't think so. As a result, why not clearly state it in the spec? It is not a big burden to clearly point it out in the spec. In addition, we already have had a subsection <No program binaries> in WebGL 2 spec, so <No shader binaries> is a counterpart in WebGL 1 spec.

File: specs/latest/1.0/index.html [hunk 2]
```diff
@@ -3481,6 +3481,14 @@ <h2 class="no-toc">Table of contents</h2>
 
 </p>
 
+    <h3>No Shader Binaries</h3>
+
+<p>
+
+Accessing binary representations of compiled shaders is not supported in the WebGL API. This includes OpenGL ES 2.0 ShaderBinary entry point. In addition, querying shader binary formats and shader compiler with getParameter is not supported in the WebGL API.
```
Comment (kenrussell) [inline specs/latest/1.0/index.html]: Can we rephrase this a little? Also, could you please wrap it to 100 characters?

"This includes OpenGL ES" -> "This includes the OpenGL ES"
"and shader compiler with" -> "and the availability of a shader compiler via"

Suggest adding:

"All WebGL implementations must implicitly support an on-line shader compiler."

Thanks.

Comment (Richard-Yunchao) [inline specs/latest/1.0/index.html]: Thanks for your review, @kenrussell . I have updated this pull request accordingly, please take another look.

File: specs/latest/1.0/index.html [hunk 3]
```diff
@@ -3481,6 +3481,21 @@ <h2 class="no-toc">Table of contents</h2>
 
 </p>
 
+    <h3>No Shader Binaries</h3>
+
+<p>
+
+Accessing binary representations of compiled shaders is not supported in the WebGL API. This
+includes the OpenGL ES 2.0 ShaderBinary entry point. In addition, querying shader binary formats
+and the availability of a shader compiler via getParameter is not supported in the WebGL API.
```
Comment (zhenyao) [inline specs/latest/1.0/index.html]: Again, please wrap <code> </code> around ShaderBinary and getParameter.

Comment (zhenyao) [inline specs/latest/1.0/index.html]: wrap "code" tag around ...

Comment (Richard-Yunchao) [inline specs/latest/1.0/index.html]: Sorry. @zhenyao, the pull request has been updated accordingly.  Please take another look.

**Comment by kenrussell** [comment]
Looks good. Thanks @Richard-Yunchao for your contribution.

**Pull request merged**